src/com/hms_networks/americas/sc/realtimedata/RealTimeTagDataPointManager.java

```diff
@@ -0,0 +1,129 @@
+package com.hms_networks.americas.sc.realtimedata;
+
+import java.util.ArrayList;
+
+import com.ewon.ewonitf.TagControl;
+import com.hms_networks.americas.sc.datapoint.DataPoint;
+import com.hms_networks.americas.sc.datapoint.DataPointBoolean;
+import com.hms_networks.americas.sc.datapoint.DataPointDword;
+import com.hms_networks.americas.sc.datapoint.DataPointFloat;
+import com.hms_networks.americas.sc.datapoint.DataPointString;
+import com.hms_networks.americas.sc.logging.Logger;
+import com.hms_networks.americas.sc.taginfo.TagInfo;
+import com.hms_networks.americas.sc.taginfo.TagType;
+
+/**
+ * This class will hold a list of data points for each tag. One instance of the class is made per
+ * tag.
+ *
+ * @author olwa
+ */
+public class RealTimeTagDataPointManager {
+  /** The name of the tag */
+  public String tagName;
+
+  /** A list of data points retrieved for the tag */
+  private ArrayList dataPoints;
+
+  /** The last data value retrieved for this tag */
+  private DataPoint lastDataPoint;
+
+  /** hold a tag control object to use when retrieving data points */
+  public TagControl tagControl;
+
+  /**
+   * Gets the tag's name
+   *
+   * @return The tag name
+   */
+  public String getTagName() {
+    return tagName;
+  }
+
+  /**
+   * Constructor
+   *
+   * @param tagName The name of the tag that data points are coming from
+   * @throws Exception when unable to create TagControl Object
+   */
+  public RealTimeTagDataPointManager(String tagName) throws Exception {
+    this.tagName = tagName;
+    this.lastDataPoint = null;
+    this.dataPoints = new ArrayList();
+    tagControl = new TagControl(tagName);
+  }
+
+  /**
+   * Add a point to the list of data points only if it is not a duplicate data point.
+   *
+   * @param data the data point to add to the data point list.
+   */
+  public void addDataPoint(DataPoint data) {
+    if (!data.equals(lastDataPoint)) {
+      dataPoints.add(data);
+      lastDataPoint = data;
+    } else {
+      Logger.LOG_INFO("duplicate data point not added. " + data.getValueString());
+    }
+  }
+
+  /**
+   * Remove from top of list
+   *
+   * @return the data point that was removed
+   */
+  public DataPoint removeDataPoint() {
+    final int lastDataPointIndex = dataPoints.size() - 1;
+    DataPoint data = (DataPoint) dataPoints.get(lastDataPointIndex);
+    dataPoints.remove(dataPoints.size() - 1);

```

Comment:
why not use lastDataPointIndex here?

Reply:
missed when incorporating change suggestions. fixed